src/Compiler/Checking/CheckExpressions.fs

```diff
@@ -729,9 +729,15 @@ let ReportImplicitlyIgnoredBoolExpression denv m ty expr =
         extractNext inner
     | expr -> checkExpr m expr
 
-let UnifyUnitType (cenv: cenv) (env: TcEnv) m ty expr =
+let UnifyUnitType (cenv: cenv) (env: TcEnv) ty expr =
     let g = cenv.g
     let denv = env.DisplayEnv
+    let rec getInnerRange expr =
+        match expr with
+        | Expr.Sequential(expr2 = expr2)  -> getInnerRange expr2
+        | expr -> expr.Range
+        
+    let m = getInnerRange expr
```

Comment:
Hmm, I guess I was thinking that we would call this (and traverse the potentially long chain of sequential expressions) only in the `else` branch below, once we knew that the overall expression's type was not `unit`, since it's specifically for the implicit discard warning that we want the more targeted range.